server/game/gameSystems/DiscardCardsFromHand.ts

```diff
@@ -0,0 +1,102 @@
+import { CardTypeFilter, EventName, GameStateChangeRequired, Location, RelativePlayer, TargetMode } from '../core/Constants';
+import { AbilityContext } from '../core/ability/AbilityContext';
+import type Player from '../core/Player';
+import { IPlayerTargetSystemProperties, PlayerTargetSystem } from '../core/gameSystem/PlayerTargetSystem';
+import { Card } from '../core/card/Card';
+import { DiscardSpecificCardSystem } from './DiscardSpecificCardSystem';
+import { cardTypeMatches } from '../core/utils/EnumHelpers';
+import { asArray } from '../core/utils/Helpers';
+import { assertTrue } from '../core/utils/Contract';
+
+export interface IDiscardCardsFromHandProperties extends IPlayerTargetSystemProperties {
+    amount: number;
+
+    /* TODO: Add reveal system to when card type filter or card condition exists, as this is required to keep the
+    in order to keep the player honest in a in-person game */
+    cardTypeFilter?: CardTypeFilter | CardTypeFilter[];
+    cardCondition?: (card: Card, context: AbilityContext) => boolean;
+}
+
+export class DiscardCardsFromHand<TContext extends AbilityContext = AbilityContext> extends PlayerTargetSystem<TContext, IDiscardCardsFromHandProperties> {
+    protected override defaultProperties: IDiscardCardsFromHandProperties = {
+        amount: 1,
+        cardCondition: () => true
+    };
+
+    public override name = 'discard';
+    public override eventName = EventName.OnCardsDiscardedFromHand;
+
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    public override eventHandler(_event): void { }
+
+    public override getEffectMessage(context: TContext): [string, any[]] {
+        const properties = this.generatePropertiesFromContext(context);
+        return ['make {0} discard {1} cards', [properties.target, properties.amount]];
+    }
+
+    public override canAffect(playerOrPlayers: Player | Player[], context: TContext, additionalProperties = {}, mustChangeGameState = GameStateChangeRequired.None): boolean {
+        for (const player of asArray(playerOrPlayers)) {
```

Comment:
nit: for helper functions like this our convention so far is to import the namespace so we can use C# / Java-style import naming. In this case, `Helper.asArray`. Same for the methods out of Contract and EnumHelpers. It just makes it clearer that it's an import and not a local method imo.